### PR TITLE
Use the UserCountry plugin as base plugin

### PR DESCRIPTION
**Pull request by ThaDafinser**
Inspired by #9 i checked the default Piwik GeoIP providers and noticted, that i can easily integrate my plugin as a provider.

Following arguments are :+1:  for this change:
- the plugin is listed and checked in `Settings -> Geolocation`
- easiert to understand for users
- we can take advantage of CLI cmd(s) and maybe other things in the future http://piwik.org/faq/how-to/faq_167/

:-1:  for this change
- how we can handle mixed enviroments (internet and intranet)? Until yet you could activate a Provider from piwik and if this plugin matched the IP with a given intranet range it was just overwritten...
- there should be no other BC break or problem...

@mattab maybe you have an idea for the mixed enviroments?

### CODE REVIEW

File: Provider.php [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+     * @return array Contains the guessed country code mapped to LocationProvider::COUNTRY_CODE_KEY.
+     */
+    public function getLocation($info)
+    {
+        // @todo this is only for testing!
+        $location = array(
+            parent::COUNTRY_CODE_KEY => 'at',
+            parent::ISP_KEY => 'github.com',
+            parent::ORG_KEY => 'OpenSource'
+        );
```
Comment (ThaDafinser) [inline Provider.php]: We need to change this, to a real detection

**Comment by mattab** [comment]
Hi @ThaDafinser 

it's quite interesting. At this stage, I don't have particular feedback: if you can create a provider in the way you need and it works, then it should be a good solution. If it's not possible yet to do what you want, maybe we could modify Piwik to allow this (or feel free to create pull request).

**Comment by ThaDafinser** [comment]
@mattab i found now the right thing for this job: Use the chain provider by Geocoder https://github.com/geocoder-php/Geocoder#the-chain-provider

This would also give Piwik some new Providers without much work. Maybe this will be a target for 3.0


Reference https://github.com/piwik/piwik/issues/6433

**Comment by Forsskieken** [comment]
Hi There,

Is there still worked on this solution?

**Comment by ThaDafinser** [comment]
@Forsskieken currently no. 
The `GeoIp` plugin from Piwik is a bit complicated...because it was not seperated properly, when it was created.

If the Plugin would be only about tracking the values, it could be easily changed, but it's also about
- update database
- viewing

I've create a `ChainProvider` for `UserAgentParsing` in the meantime but had not found time, to implement it for Piwik https://github.com/ThaDafinser/UserAgentParser

**Pull request closed without merging**